docs/book/src/developer/architecture/controllers/tuning.md

```diff
@@ -0,0 +1,122 @@
+
+Please reach out to maintainers if you are interested in helping us to make progress in this area.
+
+</aside>
+
+## Analyzing metrics, traces and profiles
+
+Tuning controllers and finding performance bottlenecks can vary depending on the issues you are dealing with, so please consider following guidelines as collection of suggestions, not as a strict process to follow.
+
+Before looking at data, it usually helps to have a clear understanding of:
+
+- What are the requirement and the constraint of the use case you are looking at, e.g.:
+  - Use a management cluster with X cpu, Y memory
+  - Create X cluster, with concurrency Y
+  - Each cluster must have X CP nodes, Y workers
+
+- What does it mean for you if the system is working well, e.g.:
+  - All machines should be provisioned in less than X minutes
+  - All controllers should reconcile in less than Y ms
+  - All controllers should allocate less than Z Gb memory
+
+Once you know the scenario you are looking at and what you are tuning for, you can finally look at data, but given that the amount of data available could be overwhelming, you probably need a strategy to navigate all the available metrics, traces, etc. .
+
+Among the many possible strategies, one usually very effective is to look at the KPIs you are aiming for, and then, if the current system performance is not good enough, start looking at other metrics trying to identify the biggest factor that is impacting the results. Usually by removing a single, performance bottleneck the behaviour of the system changes in a significant way; after that you can decide if the performance is now good enough or you need another round of tuning.
+
+Let's try to make this more clear by using an example, *machine provisioning time is degrading when running CAPI at scale* (machine provisioning time can be seen in the [Cluster API Performance dashboard](http://localhost:3001/d/b2660352-4f3c-4024-837c-393d901e6981/cluster-api-performance?orgId=1)).
+
+When running at scale, one of the first things to take care of is the client-go rate limiting, which is a mechanism built inside client-go that prevents a Kubernetes client from being accidentally too aggressive to the API server.
+However this mechanism can also limit the performance of a controller when it actually requires to make many calls to the API server.
+
+So one of the first data point to look at is the rate limiting metrics; given that upstream CR doesn't have metric for that we can only look for logs containing "client-side throttling" via http://localhost:3001/explore?orgId=1&left=%7B%22datasource%22:%22loki%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki%22%7D,%22editorMode%22:%22builder%22,%22expr%22:%22%7Bapp%3D~%5C%22capi-.%2A%5C%22%7D%20%7C%3D%20%60client-side%20throttling%60%22,%22queryType%22:%22range%22%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D)
```

Comment:
```suggestion
So one of the first data point to look at is the rate limiting metrics; given that upstream CR doesn't have metric for that we can only look for logs containing "client-side throttling" via [Loki](http://localhost:3001/explore)
```

I know the suggestion came for me, but maybe let's play it safe with the sorter link